bigframes/ml/llm.py

```diff
@@ -1018,15 +1029,34 @@ def predict(
             "ground_with_google_search": ground_with_google_search,
         }
 
-        df = self._bqml_model.generate_text(X, options)
+        df_result = bpd.DataFrame(session=self._bqml_model.session)  # placeholder
+        df_fail = X
+        for _ in range(retry + 1):
+            df = self._bqml_model.generate_text(df_fail, options)
 
-        if (df[_ML_GENERATE_TEXT_STATUS] != "").any():
+            df_succ = df[df[_ML_GENERATE_TEXT_STATUS].str.len() == 0]
+            df_fail = df[df[_ML_GENERATE_TEXT_STATUS].str.len() > 0]
```

Comment:
Can we just use df_succ to tell whether there is any failed row? Then it can reduce one filter op.

Reply:
not really. We need df_fail for next retry, and append df_succ to df_result of current round. And I don't think DF subtract will be more effecient.